### PR TITLE
Bind correct default resource type in GI

### PR DESCRIPTION
**Pull request by RandomShaper**
This is to avoid that, when `rendering/reflections/sky_reflections/texture_array_reflections` is enabled, a wrong resource type is bound, something that the Vulkan validation layers complain about:
```
drivers\vulkan\vulkan_context.cpp:159 - VALIDATION - Message Id Number: -439258052 | Message Id Name: VUID-vkCmdDispatch-None-02699
	Validation Error: [ VUID-vkCmdDispatch-None-02699 ] Object 0: handle = 0x7c0376000000040c, type = VK_OBJECT_TYPE_DESCRIPTOR_SET; | MessageID = 0xe5d1743c | Descriptor set VkDescriptorSet 0x7c0376000000040c[] encountered the following validation error at vkCmdDispatch time: Descriptor in binding #0 index 0 requires an image view of type VK_IMAGE_VIEW_TYPE_CUBE_ARRAY but got VK_IMAGE_VIEW_TYPE_CUBE. The Vulkan spec states: Descriptors in each bound descriptor set, specified via vkCmdBindDescriptorSets, must be valid if they are statically used by the VkPipeline bound to the pipeline bind point used by this command (https://vulkan.lunarg.com/doc/view/1.3.211.0/windows/1.3-extensions/vkspec.html#VUID-vkCmdDispatch-None-02699)
	Objects - 1
		Object[0] - VK_OBJECT_TYPE_DESCRIPTOR_SET, Handle 8936115828005274636
```

### CODE REVIEW

File: servers/rendering/renderer_rd/environment/gi.cpp [hunk 1]
```diff
@@ -3322,7 +3322,11 @@ void GI::init(RendererSceneSkyRD *p_sky) {
 				RD::Uniform u;
 				u.uniform_type = RD::UNIFORM_TYPE_TEXTURE;
 				u.binding = 0;
-				u.append_id(texture_storage->texture_rd_get_default(RendererRD::DEFAULT_RD_TEXTURE_CUBEMAP_WHITE));
+				if (p_sky->sky_use_cubemap_array) {
+					u.append_id(texture_storage->texture_rd_get_default(RendererRD::DEFAULT_RD_TEXTURE_CUBEMAP_ARRAY_BLACK));
```
Comment (akien-mga) [inline servers/rendering/renderer_rd/environment/gi.cpp:3326]: You implemented `DEFAULT_RD_TEXTURE_CUBEMAP_ARRAY_WHITE` but are using `DEFAULT_RD_TEXTURE_CUBEMAP_ARRAY_BLACK`, is that intended?

Comment (RandomShaper) [reply servers/rendering/renderer_rd/environment/gi.cpp:3326]: Nope... Copy-paste mistake. Fixed.

File: servers/rendering/renderer_rd/storage_rd/texture_storage.cpp [hunk 2]
```diff
@@ -219,6 +219,34 @@ TextureStorage::TextureStorage() {
 		}
 	}
 
+	{ //create default cubemap
```
Comment (akien-mga) [inline servers/rendering/renderer_rd/storage_rd/texture_storage.cpp:222]:
```suggestion
	{ //create default cubemap array
```
like for `ARRAY_BLACK`, no?

Comment (akien-mga) [reply servers/rendering/renderer_rd/storage_rd/texture_storage.cpp:222]: Or well, the comments seem backward between the scopes that create `DEFAULT_RD_TEXTURE_CUBEMAP_ARRAY_BLACK` and `DEFAULT_RD_TEXTURE_CUBEMAP_ARRAY_WHITE` (says "default cubemap") and the ones that create `DEFAULT_RD_TEXTURE_CUBEMAP_BLACK` ("default cubemap array") and `DEFAULT_RD_TEXTURE_CUBEMAP_WHITE` ("default cubemap white array").

Might be worth cleaning up to make the comments more accurate (and maybe fix capitalization / punctuation while at it).

**Review by clayjohn — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**